src/ol_infrastructure/components/services/k8s.py

```diff
@@ -0,0 +1,311 @@
+# ruff: noqa: ERA001, C416
+"""
+This is a service components that replaces a number of "boilerplate" kubernetes
+calls we currently make into one convenient callable package.
+"""
+
+from pathlib import Path
+from typing import Optional
+
+import pulumi_kubernetes as kubernetes
+from pulumi import ComponentResource, ResourceOptions
+from pydantic import BaseModel
+
+from bridge.lib.magic_numbers import DEFAULT_NGINX_PORT, DEFAULT_UWSGI_PORT
+from bridge.settings.github.team_members import DEVOPS_MIT
+from ol_infrastructure.lib.pulumi_helper import parse_stack
+
+
+class OLApplicationK8sConfiguration(BaseModel):
+    project_root: Path
+    application_config: dict[str, str]
+    application_name: str
+    application_namespace: str
+    application_lb_service_name: str
+    application_lb_service_port_name: str
+    k8s_global_labels: dict[str, str]
+    db_creds_secret_name: str
+    redis_creds_secret_name: str
+    static_secrets_name: str
+    application_security_group_id: str
+    application_docker_tag: str | None
+    vault_k8s_resource_auth_name: str
+    import_nging_config: bool
+
+
+stack_info = parse_stack()
+env_name = f"{stack_info.env_prefix}-{stack_info.env_suffix}"
+
+
+class OLApplicationK8s(ComponentResource):
+    """
+    Main K8s component resource class
+    """
+
+    def __init__(
+        self,
+        ol_app_k8s_config: OLApplicationK8sConfiguration,
+        opts: Optional[ResourceOptions] = None,
+    ):
+        """
+        It's .. the constructor. Shaddap Ruff :)
+        """
+        super().__init__(
+            "ol:infrastructure:components:services:OLApplicationK8s",
+            ol_app_k8s_config.application_namespace,
+            None,
+            opts=opts,
+        )
+        resource_options = ResourceOptions(parent=self).merge(opts)
+        self.ol_app_k8s_config: OLApplicationK8sConfiguration = ol_app_k8s_config
+
+        if ol_app_k8s_config.import_nging_config:
```

Comment:
I notice none of our other k8s apps import an nginx configuration, so I conditionalized this blob with a boolean in my configuration class.

Let me know if you think that doesn't make sense or know of a better/cleaner way :)